plugins/test/fqdn_valid.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2023, Vladimir Botka <vbotka@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.errors import AnsibleError
+from ansible.module_utils.six import raise_from
+
+try:
+    from fqdn import FQDN
+except ImportError as imp_exc:
+    ANOTHER_LIBRARY_IMPORT_ERROR = imp_exc
+else:
+    ANOTHER_LIBRARY_IMPORT_ERROR = None
+
+
+DOCUMENTATION = '''
+  name: fqdn_valid
+  short_description: Validates fully-qualified domain names against RFC 1123.
+  version_added: 8.1.0
+  author: Vladimir Botka (@vbotka)
+  requirements:
+  - fqdn>=1.5.1 (PyPI)
+  description:
+    - This test validates Fully Qualified Domain Names (FQDNs)
+      conforming to the Internet Engineering Task Force specification
+      RFC 1123 and RFC 952
```

Comment:
```suggestion
      RFC 1123 and RFC 952.
```
Or let's better link the RFCs:
```suggestion
      L(RFC 1123, https://www.rfc-editor.org/rfc/rfc1123) and L(RFC 952, https://www.rfc-editor.org/rfc/rfc952).
```